# Virus clade comparator
To identify mutations in the dataset, we developed a custom script. This script imports necessary modules, including os for interacting with the operating system, Bio.Align for sequence alignment, and Bio.SeqIO for reading and writing sequence files. It defines a function called pairwise compare that takes two sequences (a reference and a sample) as input, aligns them using the BLAST algorithm (scoring="blastn"), and compares them base by base. The script identifies differences between the sequences, including insertions, deletions, or substitutions, and records these differences in a list. It then sets the path to the working directory and reads the reference sequence from a FASTA file. Subsequently, the program parses a FASTA file containing multiple sequences for comparison (clade.fasta). The sequences are converted to uppercase, and each sequence is iterated through, compared with the reference sequence using the pairwise compare function, and stored in a list called the intersection list. The intersection list is then iterated through, and the intersection of the comparison results for each sequence is calculated. Finally, the intersection of all the comparison results is printed.
# Requirements
To run this script you need to install Python v.3.6+ and BioPython v.1.81+.
